src/library_fs.js

```diff
@@ -1073,10 +1073,13 @@ FS.staticInit();` +
       var stream = FS.createStream({
         node: node,
         path: FS.getPath(node),  // we want the absolute path to the node
+        id: node.id,
         flags: flags,
+        mode: node.mode,
         seekable: true,
         position: 0,
         stream_ops: node.stream_ops,
+        node_ops: node.node_ops,
```

Comment:
I don't see these new members being used elsewhere in this PR.. are they needed for some reason?

Reply:
hm, I'll track these down tomorrow. The [commit](https://github.com/emscripten-core/emscripten/pull/10162/commits/8b03db62ada4fa699cbb8856be0f8dee067cb543) says
> Implemented missing pieces for memory FS to readdir

suggesting it's the change @kripken made in `library_syscall.js` to `FS.lookupNode(parent, name)`, which does call things that expect `parent` to have a `.node_ops` property at least. Not sure about the others yet.

@AlexanderOMara happen to remember? Sorry both, I shouldn't have squashed these.

Reply:
Indeed it looks like the change in `__sys_getdents64()` in library_syscall.js from
```
var child = FS.lookupNode(stream.node, name);
```
to
```
var child = FS.lookupNode(stream, name);
```
makes this stream need at least some of these properties.

Reply:
Sounds good.